Building issue with  Boost >= 1.60.0 and CUDA != 7.5

Compiling with gcc 5.4.0 and CUDA 8.0 on ubuntu 16.04 the following error is thrown

```
common/include/boost/chrono/duration.hpp(256): error: class "boost::common_type<int_least64_t, int_least64_t, intmax_t>" has no member "type"
          detected during:
            instantiation of "ToDuration boost::chrono::detail::duration_cast_aux<FromDuration, ToDuration, Period, true, false>::operator()(const FromDuration &) const [with FromDuration=boost::chrono::duration<int_least64_t, boost::nano>, ToDuration=boost::chrono::seconds, Period=boost::ratio<1L, 1000000000L>]" 
(315): here
            instantiation of "ToDuration boost::chrono::detail::duration_cast<FromDuration, ToDuration>::operator()(const FromDuration &) const [with FromDuration=boost::chrono::duration<int_least64_t, boost::nano>, ToDuration=boost::chrono::seconds]" 
(789): here
            instantiation of "boost::enable_if<boost::chrono::detail::is_duration<ToDuration>, ToDuration>::type boost::chrono::duration_cast<ToDuration,Rep,Period>(const boost::chrono::duration<Rep, Period> &) [with ToDuration=boost::chrono::seconds, Rep=int_least64_t, Period=boost::nano]" 
common/include/boost/thread/pthread/timespec.hpp(51): here

common/include/boost/chrono/duration.hpp(405): error: class "boost::common_type<long, long>" has no member "type"
          detected during instantiation of class "boost::common_type<boost::chrono::duration<Rep1, Period1>, boost::chrono::duration<Rep2, Period2>> [with Rep1=int_least64_t, Period1=boost::nano, Rep2=int_least64_t, Period2=boost::ratio<1L, 1L>]" 
common/include/boost/thread/pthread/timespec.hpp(52): here

2 errors detected in the compilation of "/tmp/tmpxft_000048f0_00000000-11_cmp_list.compute_52.cpp1.ii".
CMake Error at CCTag_generated_cmp_list.cu.o.Release.cmake:279 (message):
  Error generating file
  /alicevision/CCTag/build/src/CMakeFiles/CCTag.dir/./cctag/cuda/./CCTag_generated_cmp_list.cu.o
```

Apparently this is related to a template aliases problem in boost with CUDA. 
https://github.com/ComputationalRadiationPhysics/picongpu/pull/1603

Applying the same fix as in https://github.com/ComputationalRadiationPhysics/picongpu/pull/1603 it seems to work

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,6 +138,18 @@ if(CCTAG_WITH_CUDA)
   # set(CUDA_NVCC_FLAGS         "${CUDA_NVCC_FLAGS};-gencode=arch=compute_52,code=sm_52;-gencode=arch=compute_52,code=compute_52;--default-stream;per-thread;-Xptxas;--warn-on-local-memory-usage;-Xptxas;--warn-on-spills")
   cuda_find_library_local_first(CUDA_CUDADEVRT_LIBRARY cudadevrt "\"cudadevrt\" library")
 
+  # Boost >= 1.60.0 and CUDA != 7.5 failed when used with C++11
+  # seen with boost 1.60.0 - 1.62.0 (atm latest) and CUDA 7.0, 8.0 (atm latest)
+  # CUDA 7.5 works without a workaround
+  # from https://github.com/ComputationalRadiationPhysics/picongpu/pull/1603
+  if( (Boost_VERSION GREATER 105999) AND
+      (NOT CUDA_VERSION VERSION_EQUAL 7.5) AND
+      (NOT CMAKE_CXX_STANDARD EQUAL 98) )
+      # Boost Bug https://svn.boost.org/trac/boost/ticket/11897
+      message(STATUS "Boost: Disable template aliases")
+      add_definitions(-DBOOST_NO_CXX11_TEMPLATE_ALIASES)
+  endif()
+
 else(CCTAG_WITH_CUDA)
   message( STATUS "Building without CUDA" )
 endif(CCTAG_WITH_CUDA)

